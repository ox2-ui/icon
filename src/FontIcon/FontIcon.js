import PropTypes from 'prop-types';
import React from 'react';
import classNames from 'classnames';

/**
 * FontIcon Component
 */
const FontIcon = props => {
  const {
    className: classNameProp,
    color,
    group,
    icon,
    size,
    style,
    ...other
  } = props;

  const className = classNames(
    `icon-${group}\:${icon}`,
    classNameProp,
  );

  const mergedStyles = {
    ...{ fontSize: size, color: color },
    ...style,
  };

  return (
    <span
      className={className}
      style={mergedStyles}
      {...other}
    />
  );
};

FontIcon.propTypes = {
  /**
   * The css class name of the root element.
   */
  className: PropTypes.string,
  /**
   * Color
   */
  color: PropTypes.string,
  /**
   * Icon group
   */
  group: PropTypes.string.isRequired,
  /**
   * Icon name
   */
  icon: PropTypes.string.isRequired,
  /**
   * Size
   */
  size: PropTypes.string,
  /**
   * Override the inline-styles of the root element.
   */
  style: PropTypes.object,
};

export default FontIcon;
