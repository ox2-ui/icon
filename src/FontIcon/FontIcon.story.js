import React from 'react';
import { storiesOf } from '@kadira/storybook';
import FontIcon from './FontIcon';

storiesOf('FontIcon', module)
  .add('default', () => (
    <FontIcon group="module" icon="users2" />
  ))
  .add('colored', () => (
    <FontIcon
      color="#2CAF65"
      group="module"
      icon="users2"
    />
  ))
  .add('colored & large', () => (
    <FontIcon
      color="#2CAF65"
      group="module"
      icon="users2"
      size="35px"
    />
  ))
  .add('ui group', () => (
    <FontIcon
      color="#2CAF65"
      group="ui"
      icon="cross"
      size="35px"
    />
  ))
  .add('style object', () => (
    <FontIcon
      color="#FFFFFF"
      group="module"
      icon="users2"
      size="35px"
      style={{
        backgroundColor: 'mediumvioletred',
        padding: '10px',
      }}
    />
  ))
  .add('themed', () => (
    <FontIcon
      className="text-color:twitter"
      group="module"
      icon="users2"
      size="35px"
    />
  ));
