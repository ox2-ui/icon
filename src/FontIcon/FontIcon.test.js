/* eslint-disable no-unused-vars */
import React from 'react';
import { shallow, render, mount } from 'enzyme';
import FontIcon from './FontIcon';

const handler = () => true;

it('renders correctly', () => {
  const wrapper = mount(
    <FontIcon
      color="#FFFFFF"
      icon="users2"
      size="35px"
      style={{ backgroundColor: 'mediumvioletred', padding: '10px' }}
    />,
  );

  expect(wrapper).toMatchSnapshot();
});
