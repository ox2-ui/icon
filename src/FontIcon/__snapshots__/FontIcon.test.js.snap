// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<FontIcon
  className=""
  color="#FFFFFF"
  group="module"
  icon="users2"
  size="35px"
  style={
    Object {
      "backgroundColor": "mediumvioletred",
      "padding": "10px",
    }
  }
>
  <span
    className="icon-module:users2"
    style={
      Object {
        "backgroundColor": "mediumvioletred",
        "color": "#FFFFFF",
        "fontSize": "35px",
        "padding": "10px",
      }
    }
  />
</FontIcon>
`;
